cpg-core/src/main/java/de/fraunhofer/aisec/cpg/frontends/LanguageTraits.kt

```diff
@@ -144,3 +144,14 @@ interface HasElaboratedTypeSpecifier : LanguageTrait {
 interface HasUnknownType : LanguageTrait {
     val unknownTypeString: List<String>
 }
+
+/**
+ * A language trait, that specifies that this language has binary operators that will short-circuit
+ * evaluation if the logical result is already known: '&&', '||' in Java or 'and','or' in Python
+ */
+interface HasShortCircuitOperators : LanguageTrait {
+    // '&&', 'and', '^'
+    val conjunctiveOperators: List<String>
```

Comment:
We have to differentiate to give the EOG property edges the right `branch` literal: `true` or `false`.
While we could make that trait an interface of every language or even make it a bunch of members instead of an interface, this has the drawback that there are several language that do not have such a feature. However, those frontends could explicitly empty the arrays. I like it more to model it as a trait as it represents what is the best.